compiler-rt/lib/scudo/standalone/combined.h

```diff
@@ -1256,7 +1275,8 @@ class Allocator {
   }
 
   void storePrimaryAllocationStackMaybe(const Options &Options, void *Ptr) {
-    if (!UNLIKELY(Options.get(OptionBit::TrackAllocationStacks)))
+    if (!UNLIKELY(Options.get(OptionBit::TrackAllocationStacks)) ||
+        !RawRingBuffer)
```

Comment:
Now the logic of enabling allocation stack becomes slightly complicated. Can we make this logic a function?

Reply:
I don't think that is a good idea. In a follow up I make the RingBuffer atomic, so we need to load it and keep the pointer.

Reply:
I don't understand why the follow up forbids packing the logic to a function?

It can be something like `getRingBuffer` and only returns non-nullptr when it's enabled.

Reply:
Sorry, I did this change but messed up a rebase, I'll upload it again shortly